From 2f4ad5e2dca9144fb11d6bf0b8fdc8a3bbdbf12f Mon Sep 17 00:00:00 2001
From: Janne Grunau <j@jannau.net>
Date: Sat, 10 Jan 2026 14:56:48 +0100
Subject: [PATCH 2/2] fixup! iio: common: Add AOP sensor drivers

Signed-off-by: Janne Grunau <j@jannau.net>
---
 drivers/iio/common/aop_sensors/Kconfig | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/iio/common/aop_sensors/Kconfig b/drivers/iio/common/aop_sensors/Kconfig
index 2c092e63a9588..6f6949bdf6a5d 100644
--- a/drivers/iio/common/aop_sensors/Kconfig
+++ b/drivers/iio/common/aop_sensors/Kconfig
@@ -15,6 +15,7 @@ config IIO_AOP_SENSOR_ALS
 	depends on ARCH_APPLE || COMPILE_TEST
 	depends on RUST
 	depends on SYSFS
+	depends on RUST_FW_LOADER_ABSTRACTIONS
 	select APPLE_AOP
 	help
 	  Module to handle the ambient light sensor attached to the AOP
-- 
2.52.0

