diff --git a/systemd/pinenote-hrdl-extract-waveform.service.orig b/systemd/pinenote-hrdl-extract-waveform.service
index dddbc1e..0aa59b2 100644
--- a/systemd/pinenote-hrdl-extract-waveform.service.orig
+++ b/systemd/pinenote-hrdl-extract-waveform.service
@@ -3,8 +3,8 @@ Description=Extract rockchip_ebc ebc.wbf waveform from waveform partition
 
 [Service]
 Type=oneshot
-ExecCondition=/usr/bin/test ! -e /usr/lib/firmware/rockchip/ebc.wbf
-ExecStart=/usr/bin/waveform_extract.sh
+ExecCondition=/usr/bin/test ! -s /usr/lib/firmware/rockchip/ebc.wbf
+ExecStart=/usr/lib/pinenote/waveform_extract.sh
 
 [Install]
 WantedBy=default.target

diff --git a/systemd/pinenote-hrdl-convert-waveform.service.orig b/systemd/pinenote-hrdl-convert-waveform.service
index b57b57e..4a5087c 100644
--- a/systemd/pinenote-hrdl-convert-waveform.service.orig
+++ b/systemd/pinenote-hrdl-convert-waveform.service
@@ -5,9 +5,9 @@ Before=display-manager.service
 
 [Service]
 Type=oneshot
-ExecCondition=/usr/bin/test ! -e /usr/lib/firmware/rockchip/custom_wf.bin
-ExecCondition=/usr/bin/test -e /usr/lib/firmware/rockchip/ebc.wbf
-ExecStart=/bin/sh -c 'cd /tmp && wbf_to_custom.py /usr/lib/firmware/rockchip/ebc.wbf && mv custom_wf.bin /usr/lib/firmware/rockchip/custom_wf.bin && mkinitcpio -P && (modprobe -r rockchip_ebc; modprobe rockchip_ebc)'
+ExecCondition=/usr/bin/test ! -s /usr/lib/firmware/rockchip/custom_wf.bin
+ExecCondition=/usr/bin/test -s /usr/lib/firmware/rockchip/ebc.wbf
+ExecStart=/bin/sh -c 'cd /tmp && /usr/lib/pinenote/wbf_to_custom.py /usr/lib/firmware/rockchip/ebc.wbf && mv custom_wf.bin /usr/lib/firmware/rockchip/custom_wf.bin && mkinitfs && (modprobe -r rockchip_ebc; modprobe rockchip_ebc)'
 
 [Install]
 RequiredBy=display-manager.service
